From: Guido Trotter <ultrotter@google.com>
To: Ganeti Development <ganeti-devel@googlegroups.com>
Subject: [PATCH stable-2.5] KVM: support version reported by 1.0
Date: Thu,  5 Jan 2012 16:22:36 +0000

This of course was working for all the rcs, but broke with 1.0 itself.

Signed-off-by: Guido Trotter <ultrotter@google.com>
---
 lib/hypervisor/hv_kvm.py |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

--- a/lib/hypervisor/hv_kvm.py
+++ b/lib/hypervisor/hv_kvm.py
@@ -199,7 +199,7 @@
   _MIGRATION_INFO_MAX_BAD_ANSWERS = 5
   _MIGRATION_INFO_RETRY_DELAY = 2
 
-  _VERSION_RE = re.compile(r"\b(\d+)\.(\d+)\.(\d+)\b")
+  _VERSION_RE = re.compile(r"\b(\d+)\.(\d+)(\.(\d+))?\b")
 
   ANCILLARY_FILES = [
     _KVM_NETWORK_SCRIPT,
@@ -937,8 +937,14 @@
     if not match:
       return None
 
-    return (match.group(0), int(match.group(1)), int(match.group(2)),
-            int(match.group(3)))
+    v_all = match.group(0)
+    v_maj = int(match.group(1))
+    v_min = int(match.group(2))
+    if match.group(4):
+      v_rev = int(match.group(4))
+    else:
+      v_rev = 0
+    return (v_all, v_maj, v_min, v_rev)
 
   def StopInstance(self, instance, force=False, retry=False, name=None):
     """Stop an instance.
